tests/e2e/scale/test_osd_node_balancing.py

```diff
@@ -0,0 +1,247 @@
+"""
+Test osd node balancing by adding nodes and osds and checking their distribution
+"""
+import logging
+from functools import reduce
+from uuid import uuid4
+from ocs_ci.framework.testlib import scale_changed_layout
+from ocs_ci.ocs import constants, defaults, machine, ocp
+from ocs_ci.ocs.exceptions import (
+    CephDetectVersionPodFound,
+    NodesNotReadyYet,
+)
+from ocs_ci.ocs.node import (
+    add_new_node_and_label_it,
+    get_osd_running_nodes,
+    get_nodes,
+    get_node_objs,
+    wait_for_nodes_status,
+)
+from ocs_ci.ocs.perfresult import PerfResult
+from ocs_ci.ocs.perftests import PASTest
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.ocs.resources.pod import get_all_pods
+from ocs_ci.utility import templating
+from ocs_ci.utility.retry import retry
+from ocs_ci.utility.utils import ceph_health_check, wait_for_cephfs_to_finish
+
+FINAL_REPORT = "Final Report"
+INITIAL_SETUP = "Initial Setup"
+MAX_NODE_COUNT = 9
+MAX_OSDS_PER_NODE = 3
+START_NODE_NUM = 3
+REPLICA_COUNT = 3
+OSD_LIMIT_AT_START = MAX_OSDS_PER_NODE * START_NODE_NUM
+MAX_TIMES_ADDED = 3
+
+
+def add_some_worker_nodes(elastic_info):
+    """
+    Add three worker nodes to the storage cluster.  Update the report file.
+
+    Args:
+        elastic_info -- ElasticData object
+    """
+    osd_running_nodes = get_osd_running_nodes()
+    logging.info(f"OSDs are running on nodes {osd_running_nodes}")
+    # Get the machine name using the node name
+    machine_names = [
+        machine.get_machine_from_node_name(osd_node) for osd_node in osd_running_nodes
+    ]
+    logging.info(f"{osd_running_nodes} associated machines are {machine_names}")
+    # Get the machineset name using machine name
+    machineset_names = [
+        machine.get_machineset_from_machine_name(machine_name)
+        for machine_name in machine_names
+    ]
+    # This next section makes new nodes on which OSD pods can run
+    for mnames in range(0, REPLICA_COUNT):
+        add_new_node_and_label_it(machineset_names[mnames], mark_for_ocs_label=True)
+    wait_for_nodes_status()
+    retry(
+        (CephDetectVersionPodFound),
+        tries=18,
+        delay=10,
+        backoff=1,
+    )(find_problem_pods)()
+    collect_stats(f"{REPLICA_COUNT} nodes have been added", elastic_info)
+
+
+def increase_osd_capacity(elastic_info):
```

Comment:
@wusui Please check function "def scale_capacity_with_deviceset(add_deviceset_count=2):" in PR #2697 for scaling cluster capacity, and use this helper function.

Reply:
This was done in my last checkin